cl/users/forms.py

```diff
@@ -188,6 +190,16 @@ def clean_email(self):
             )
         return email
 
+    def clean_first_name(self):
+        first_name = self.cleaned_data.get("first_name")
+        if not re.match(
+            r"""[^!"#$%&()*+,./:;<=>?@[\]_{|}~]+$""", first_name, re.IGNORECASE
+        ):
```

Comment:
Boy, I'm rusty on regexes these days, but why not do:

```suggestion
        import string
        if re.match(f"[{string.punctuation}]+$", first_name):
```

That flips the logic, uses `string.punctuation` (which I think is fine?) and removes `re.IGNORECASE`, because I can't see how it was relevant?

Reply:
@mlissner You're absolutely right! The re.IGNORECASE flag isn't necessary here, and your suggestion is great.

Before we implement this change, I'd like to clarify one thing:

The current regex allows names with apostrophes (') and hyphens/dashes (-). If we use `string.punctuation` to validate names, it will flag these characters as invalid. Is that ok?

Reply:
👍 :shipit:

Reply:
Wait, before we ship, we can flip the regex, remove the `not`, and remove the `IGNORECASE`, right?

Reply:
Thank you for pointing that out. I've incorporated the refactor into the latest commit